File: Coding Sample/Java/Fundamentals/Concurrency/ThreadSleepTest.java
class ThreadSleepTest{

	static class GreeterThread extends Thread{
		
		private String message;
		private int count;

		GreeterThread(String m, int n){
			message = m;
			count = n;
			start();
		}

		@Override
		public void run(){
			for(int i = 1; i <= count; ++i){
				System.out.printf("%s:%d from thread<%x>%n", message, i, Thread.currentThread().hashCode());
				Worker.doWork(i);
			}
			try{
				Thread.sleep(15000);
			}catch(InterruptedException e){
				System.out.println("Sleep interrupted!");
			}
			System.out.println("Goodbye.");
		}

	}

	public static void main(String[] args) throws Exception{
		Thread child = new GreeterThread("Hello", 8);
		for(int i = 12; i > 0; i--){
			System.out.printf("Welcome:%d from thread<%x>%n", i, Thread.currentThread().hashCode());
			Worker.doWork(i);
		}
		child.interrupt();
	}
}

File: Coding Sample/MS.NET/Fundamentals/Concurrency/threadsleeptest.cs
using System;
using System.Threading;

static class Program
{
	class GreetArg
	{
		public string Message;

		public int Count;
	}

	private static void GreetStart(object arg)
	{
		GreetArg ga = (GreetArg)arg;

		for(int i = 1; i <= ga.Count; ++i)
		{
			Console.WriteLine("{0}:{1} from thread<{2}>", ga.Message, i, Thread.CurrentThread.ManagedThreadId);
			Worker.DoWork(i);
		}

		try
		{
			Thread.Sleep(10000);
		}
		catch(ThreadInterruptedException)
		{
			Console.WriteLine("Sleep interrupted!");
		}

		Console.WriteLine("Goodbye.");
	}

	public static void Main()
	{
		Thread child = new Thread(GreetStart);
		child.Start(new GreetArg{Message="Hello", Count=10});

		for(int i = 15; i > 0; i--)
		{
			Console.WriteLine("Welcome:{0} from thread<{1}>", i, Thread.CurrentThread.ManagedThreadId);
			Worker.DoWork(i);
		}

		child.Interrupt();
	}
}
